docs/learn/guides/file-integrations/file-integration-using-ftp/README.md

```diff
@@ -0,0 +1,198 @@
+
+## Prerequisites
+ 
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install one of the following Ballerina IDE plugins: 
+>[VSCode](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina), 
+>[IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+- An FTP Server (See [here](https://www.digitalocean.com/community/tutorials/how-to-set-up-vsftpd-for-a-user-s-directory-on-ubuntu-16-04) on how to setup an FTP server)
+
+## Implementation
+> If you want to skip the basics, you can download the GitHub repo and directly move to the "Testing" section by skipping  "Implementation" section.
+
+### Create the project structure
+
+Ballerina is a complete programming language that supports custom project structures. Use the following package structure for this guide.
+```
+file-integration-using-ftp
+    ├── src
+       └── ftp_listener
+           └── ftp_listener.bal
+```
+
+Create the Ballerina project `file-integration-using-ftp` and add the `ftp_listener` module using the below commands. 
+
+```bash
+    $ ballerina new file-integration-using-ftp
+    $ cd file-integration-using-ftp
+    $ ballerina add ftp_listener
+```
+
+The above package structure will be created for you. Create the `ftp_listener.bal` file in the Ballerina module.
+
+### Developing the FTP listener service
+
+Let's start implementation by importing the WSO2 FTP Connector in the `ftp_listener.bal` file you just created.
+
+```ballerina
+import wso2/ftp;
+```
+
+Then add the configuration for file type and file location to listen to.
+
+ ```ballerina
+type Config record {
+    string fileNamePattern;
+    string filePath;
+};
+
+Config conf = {
+    fileNamePattern: config:getAsString("FTP_FILE_NAME_PATTERN"),
+    filePath: config:getAsString("FTP_LISTENER_PATH")
+};
+```
+
+Create an FTP listener instance by defining the configuration.
+
+```ballerina
+listener ftp:Listener dataFileListener = new({
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+        basicAuth: {
+            username: config:getAsString("FTP_USERNAME"),
+            password: config:getAsString("FTP_PASSWORD")
+        }
+    },
+    path: conf.filePath,
+    fileNamePattern: conf.fileNamePattern
+    pollingInterval: config:getAsInt("FTP_POLLING_INTERVAL"),
+});
+```
+
+Then implement the FTP client endpoint to do the file processing.
+     
+```ballerina
+ftp:ClientEndpointConfig ftpConfig = {
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+     basicAuth: {
+         username: config:getAsString("FTP_USERNAME"),
+         password: config:getAsString("FTP_PASSWORD")
+     }
+    }
+};
+
+ftp:Client ftpClient = new(ftpConfig);
+```
+
+Create the service to listen to the FTP server. 
```

Comment:
Add some detail about added files and deleted files.

Reply:
Added.